dbt/adapters/default/impl.py

```diff
@@ -162,7 +162,8 @@ def drop_relation(cls, profile, project_cfg, relation, model_name=None):
 
         sql = 'drop {} if exists {} cascade'.format(relation.type, relation)
 
-        connection, cursor = cls.add_query(profile, sql, model_name)
+        connection, cursor = cls.add_query(profile, sql, model_name,
```

Comment:
I had to make this change because dbt was previously adding a `begin` statement before the `drop`. Since the `drop` now happens _after_ the transaction, there was no corresponding `commit`, and the transaction containing the `drop` was rolled back.

I checked the other places where dbt uses this method, and I believe this change should be safe, but I'm very open to improving how dbt handles transactions here